design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,190 @@
+# Rook-Ceph Bucket Provisioning
+
+## Proposed Feature
+
+Add a CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph operator for the dynamic provisioning of S3 buckets by Kubernetes users.
+
+### Overview
+
+  Rook-Ceph has developed Custom Resource Definitions (CRDs) to support the dynamic provisioning of Ceph Object Stores and Ceph Object Store Users.  This proposal introduces the next logical step: dynamic bucket provisioning with the addition of a CephObjectBucketClaim CRD.  The intent is to round out the Rook-Ceph experience for Kubernetes users by providing a single control plane for the managing of Ceph Object components.      
+
+## Goals
+
+- Provide Rook-Ceph users the ability to dynamically provision object buckets via the Kubernetes API with a CephObjectBucketClaim CRD.
+- Enable cluster admin control over bucket creation and object store access via the Kubernetes API.
+- Utilize a familiar and deterministic pattern when injecting bucket connection information into workload environments. 
+
```

Comment:
just for completeness I'd add:
`- Ensure the app pod that will reference the bucket will not run prior to the bucket being fully provisioned and secured.`

Reply:
And, based on ACL requirement below, I'd add this goal:
`- user-based credentials (Secret) cannot be usinmg outside of Kubernetes to create additional buckets`